packages/shared/src/modules/users/libs/validation-schemas/user-sign-up.validation-schema.ts

```diff
@@ -11,14 +11,29 @@ const userSignUp: z.ZodType<UserSignUpRequestDto> = z
 			.min(UserValidationRule.EMAIL_MINIMUM_LENGTH, {
 				message: UserValidationMessage.EMAIL_REQUIRED,
 			})
-			.email({
-				message: UserValidationMessage.EMAIL_WRONG,
-			}),
+			.max(UserValidationRule.EMAIL_MAXIMUM_LENGTH, {
+				message: UserValidationMessage.EMAIL_TOO_LONG,
+			})
+			.refine(
+				(email) => {
+					const [localPart, domainPart] = email.split("@");
+
+					return (
+						!!localPart &&
+						!!domainPart &&
+						UserValidationRule.EMAIL_LOCAL_PART_PATTERN.test(localPart) &&
+						UserValidationRule.EMAIL_DOMAIN_PART_PATTERN.test(domainPart)
+					);
+				},
+				{
+					message: UserValidationMessage.EMAIL_WRONG,
+				},
+			),
 		name: z
 			.string()
 			.trim()
 			.min(UserValidationRule.NAME_MINIMUM_LENGTH, {
-				message: UserValidationMessage.NAME_REQUIRED,
+				message: UserValidationMessage.NAME_TOO_SHORT,
```

Comment:
I'd propose to leave `NAME_REQUIRED`. We can only use`.min(1)`  to check required string with zod, and 'name is required' message is more propriate than 'need more than 1 symbol' or smth

Reply:
Since the name must be at least 3 characters, I think `NAME_TOO_SHORT` is a better fit than just `NAME_REQUIRED`.

Reply:
can't we pass required error additionally, like this `.string({ required_error: ''})`

Reply:
Yes, we can pass a required error using `.string({ required_error: '' })`. However, it won't be seen if the minimum length error is triggered first.
```
email: z
			.string({
				required_error: UserValidationMessage.EMAIL_REQUIRED,
			})
			.trim()
			.min(UserValidationRule.EMAIL_MINIMUM_LENGTH, {
				message: UserValidationMessage.EMAIL_TOO_SHORT,
			})
```
![image](https://github.com/user-attachments/assets/5e2aebee-a013-42a0-be87-65d1f59eec3b)

Reply:
Hm, got this. then let's leave it

Reply:
I'm thinking that maybe in this case we can change other messages with required, for example for name can: name can be at least characters long